examples/crt/main.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/png"
+	"log"
+	"math"
+	"math/rand"
+	"time"
+
+	"golang.org/x/image/font"
+	"golang.org/x/image/font/opentype"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/examples/resources/fonts"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+	"github.com/hajimehoshi/ebiten/v2/text"
+)
+
+// Game implements ebiten.Game interface.
+type Game struct {
+	Objects   []*Object
+	Offscreen *ebiten.Image
+	Time      float32
+	Score     int
+}
+
+type Name int
+
+const (
+	Player Name = iota
+	Ebiten Name = iota
+	Floor  Name = iota
+)
+
+// Object implements an object that the game loop modifies and the draw loop renders
+type Object struct {
+	Name          Name
+	X, Y, Z, W, H int
+	VX, VY        float64
+	Flipped       bool
+	Image         *ebiten.Image
+	Tile          bool
+	TileX         int
+	TileY         int
+	Hidden        bool
+}
+
+// Draw executes graphical operations to draw a scene
+func (o *Object) Draw(screen *ebiten.Image, OffsetX, OffsetY float64) {
+	if !o.Tile {
+		opts := &ebiten.DrawImageOptions{}
+		if o.Flipped {
+			opts.GeoM.Scale(-1, 1)
+			opts.GeoM.Translate(float64(o.W), 0)
+		}
+		opts.GeoM.Translate(OffsetX+float64(o.X), OffsetY+float64(o.Y))
+		screen.DrawImage(o.Image, opts)
+	} else {
+		for x := 0; x < o.TileX; x++ {
+			for y := 0; y < o.TileY; y++ {
+				opts := &ebiten.DrawImageOptions{}
+				opts.GeoM.Translate(OffsetX+float64(o.X)+float64(x*o.W), OffsetY+float64(o.Y)+float64(y*o.H))
+				screen.DrawImage(o.Image, opts)
+			}
+		}
+	}
+}
+
+func getPlayerIndex(objs []*Object) (int, error) {
+	for i, obj := range objs {
+		if obj.Name == Player {
+			return i, nil
+		}
+	}
+	return 0, errors.New("Couldn't find player object")
```

Comment:
If this should never happen, `panic` is also an option. Errors are still ignored anyway.